addons/mail/static/src/core/common/store_service.js

```diff
@@ -186,6 +186,9 @@ export class Store extends BaseStore {
         if (ev.target.closest(".o_channel_redirect") && model && id) {
             ev.preventDefault();
             const thread = this.Thread.insert({ model, id });
+            if (!thread.type) {
+                this.env.services["mail.thread"].fetchChannel(id);
```

Comment:
We might want to `await` the fetch before opening, otherwise the issue will still be present with a flicker.